File: atcoder/370/B.py
def main():
    N = int(input())
    A = []
    genso = 1
    for i in range(N):
        row = list(map(int, input().split()))
        A.append(row)
    for i in range(N):
        if genso-1 >= i:
            genso = A[genso - 1][i]
        else:
            genso = A[i][genso - 1]
    print(genso)
if __name__ == "__main__":
    main()    

File: atcoder/370/B.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
    int N;
    cin >> N;
    int genso = 1;
    vector<vector<int>> A(N, vector<int>(N));
    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j <= i; j++)
        {
            cin >> A[i][j];
        }
    }
    for (int i = 0; i < N; i++)
    {
        if (genso - 1 >= i)
        {
            genso = A[genso - 1][i];
        }
        else
        {
            genso = A[i][genso - 1];
        }
    }
    cout << genso << endl;
    return 0;
}